app/controllers/compute_resources_vms_controller.rb

```diff
@@ -115,4 +121,18 @@ def run_vm_action(action)
     error _("Error - %{message}") % { :message => _(e.message) }
     redirect_to :back
   end
+
+  def load_vms
+    opts = {}
+    if @compute_resource.supports_vms_pagination?
+      return if request.format.html? # vms are needed only for json, html loads only thead
+      opts[:search] = params['sSearch'] || ''
+      opts[:max] = (params['iDisplayLength'] || 10).to_i
+      opts[:page] = (params['iDisplayStart'].to_i / opts[:max])+1
+      opts[:without_details] = true
+    else
+      opts = params[:filters] if params[:filters].present?
```

Comment:
I mean is `params[:filters]` actually ever passed in to this controller? I could not find anywhere it is passed, and it has been here since before datatables was implemented - leading me to think this is remains from an old implementation.